src/category_theory/single_obj.lean

```diff
@@ -0,0 +1,124 @@
+import category_theory.endomorphism category_theory.groupoid category_theory.Cat
+import data.equiv.algebra algebra.Mon.basic
+import tactic.find
+
+/-!
+Copyright (c) 2019 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+
+# Single-object category
+
+Single object category with a given monoid of endomorphisms.  It is defined to faciliate transfering
+some definitions and lemmas (e.g., conjugacy etc.) from category theory to monoids and groups.
+
+## Main definitions
+
+Given a type `α` with a monoid structure, `single_obj α` is `unit` type with `category` structure
+such that `End (single_obj α).star` is the monoid `α`.  This can be extended to a functor `Mon ⥤
+Cat`.
+
+If `α` is a group, then `single_obj α` is a groupoid.
+
+An element `x : α` can be reinterpreted as an element of `End (single_obj.star α)` using
+`single_obj.to_End`.
+
+## Implementation notes
+
+- `category_struct.comp` on `End (single_obj.star α)` is `flip (*)`, not `(*)`. This way
+  multiplication on `End` agrees with the multiplication on `α`.
+
+- By default, Lean puts instances into `category_theory` namespace instead of
```

Comment:
Huh? Once you are inside the `namespace single_obj` this shouldn't happen I think.